apps/js-sdk/firecrawl/src/index.ts

```diff
@@ -942,18 +941,39 @@ interface CrawlWatcherEvents {
 }
 
 export class CrawlWatcher extends TypedEventTarget<CrawlWatcherEvents> {
-  private ws: WebSocket;
+  private ws!: WebSocket;
+  private initialized: Promise<void>;
   public data: FirecrawlDocument<undefined>[];
   public status: CrawlStatusResponse["status"];
   public id: string;
 
+  private static async loadWebSocket() {
+    try {
+      if (typeof require !== 'undefined') {
+        return require('isows').WebSocket;
+      } else {
+        const module = await import('isows');
+        return module.WebSocket;
+      }
+    } catch (e) {
+      throw new FirecrawlError('Failed to load WebSocket implementation', 500);
+    }
+  }
+
   constructor(id: string, app: FirecrawlApp) {
     super();
     this.id = id;
-    this.ws = new WebSocket(`${app.apiUrl}/v1/crawl/${id}`, app.apiKey);
     this.status = "scraping";
     this.data = [];
 
+    this.initialized = CrawlWatcher.loadWebSocket()
```

Comment:
I dont think this is good enough, cause ws is not initialized when the constructor is called but later async, so there can be a race condition.